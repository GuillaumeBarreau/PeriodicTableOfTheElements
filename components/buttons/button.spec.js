import { shallowMount } from '@vue/test-utils'
import cmp from '~/components/buttons/button.vue'

describe('testing component button', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(cmp, {
      slots: {
        default: 'slotsText test'
      },
      data: () => {
        return {
          modalIsOpen: 'false'
        }
      }
    })
  })

  it('snapshot testing', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Component Validation', () => {
    it('Should render the component button.vue', () => {
      expect(wrapper.is(cmp)).toBe(true)
    })
  })

  describe('Content validation', () => {
    it('should render the correct text', () => {
      expect(wrapper.text()).toContain('slotsText test')
    })
  })

  describe('Methods validation', () => {
    it('Should check if the method showData is called ', () => {
      const showData = jest.fn()
      wrapper.setMethods({ showData })

      const button = wrapper.find('button')
      button.trigger('click')

      expect(wrapper.vm.showData).toHaveBeenCalled()
    })
  })
})
