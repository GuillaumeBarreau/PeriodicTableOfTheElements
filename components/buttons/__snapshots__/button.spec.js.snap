// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component button snapshot testing 1`] = `"<button class=\\"content--button\\">slotsText test</button>"`;
